infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/TolerationsProvisioner.java

```diff
@@ -54,4 +55,10 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       k8sEnv.getPodsData().values().forEach(d -> d.getSpec().setTolerations(tolerations));
     }
   }
+
+  public void provision(PodSpec podSpec) {
+    if (!tolerations.isEmpty()) {
```

Comment:
The above `provision` method sets the tolerations unconditionally, even if empty. Do we need a different behavior here?

Reply:
Same reply here - I think GitHub is merely hiding the condition in this view, but both methods use the same one.